tests/devices/unit_tests/test_gridscan.py

```diff
@@ -27,49 +27,67 @@ def discard_status(st: Status | DeviceStatus):
 
 
 @pytest.fixture
-async def zebra_fast_grid_scan(request):
+async def zebra_fast_grid_scan():
     async with DeviceCollector(mock=True):
         zebra_fast_grid_scan = ZebraFastGridScan(name="fake_FGS", prefix="FGS")
 
     return zebra_fast_grid_scan
 
 
 @pytest.fixture
-async def panda_fast_grid_scan(request):
+async def panda_fast_grid_scan():
     async with DeviceCollector(mock=True):
-        panda_fast_grid_scan = PandAFastGridScan(name="fake_FGS", prefix="FGS")
+        panda_fast_grid_scan = PandAFastGridScan(name="fake_PGS", prefix="PGS")
 
     return panda_fast_grid_scan
 
 
+@pytest.mark.parametrize(
+    "use_pgs",
+    [(False), (True)],
+)
 async def test_given_settings_valid_when_kickoff_then_run_started(
+    use_pgs,
     zebra_fast_grid_scan: ZebraFastGridScan,
+    panda_fast_grid_scan: PandAFastGridScan,
 ):
-    set_mock_value(zebra_fast_grid_scan.scan_invalid, False)
-    set_mock_value(zebra_fast_grid_scan.position_counter, 0)
-    set_mock_value(zebra_fast_grid_scan.status, 1)
+    grid_scan: ZebraFastGridScan | PandAFastGridScan = (
+        panda_fast_grid_scan if use_pgs else zebra_fast_grid_scan
+    )
```

Comment:
Nit: So I think you could be [clever](https://x.com/TitusWinters/status/1143595692113481728) here with fixtures (https://engineeringfordatascience.com/posts/pytest_fixtures_with_parameterize/#using-requestgetfixturevalue-) and do something like:
```python
@pytest.mark.parametrize(
    "grid_scan",
    ["zebra_fast_grid_scan", "panda_fast_grid_scan"],
)
async def test_given_settings_valid_when_kickoff_then_run_started(
    grid_scan_type: str, request
):
    grid_scan: ZebraFastGridScan | PandAFastGridScan = request.getfixturevalue(grid_scan_type)
```

But I appreciate that this may cause more confusion/wackiness down the line so leave it up to you if you think it's nicer or not.

Reply:
I just had a play with this, I think there are some issues with using `request.getfixturevalue` with `asyncio` fixtures. There's probably ways around this, but I'll leave it for now